pkg/virt-handler/device-manager/pci_device_test.go

```diff
@@ -104,13 +110,49 @@ pciHostDevices:
 			}
 		}
 		// discoverPermittedHostPCIDevices() will walk real PCI devices wherever the tests are running
-		// It's assumed here that it will find a PCI device at 0000:00:00.0
+		// It's assumed here that it will find a PCI device
 		pciDevices := discoverPermittedHostPCIDevices(supportedPCIDeviceMap)
 		devs := constructDPIdevices(pciDevices[fakeName], iommuToPCIMap)
-		Expect(devs[0].ID).To(Equal(fakeIommuGroup))
+		Expect(devs[0].ID).To(Equal(strings.Join([]string{fakeIommuGroup, fakeAddress}, deviceIDSeparator)))
 		Expect(devs[0].Topology.Nodes[0].ID).To(Equal(int64(fakeNumaNode)))
 	})
+
+	It("Should validate multiple devices in IOMMU group", func() {
+		By("making sure the environment has at least 2 PCI devices")
+		if len(pciAddresses) < 2 {
+			Skip("the second PCI device was not found")
+		}
+		secondFakeAddress := pciAddresses[1]
+		By("mocking PCI functions to simulate a vfio-pci device at " + secondFakeAddress)
+		mockPCI.EXPECT().GetDeviceIOMMUGroup(pciBasePath, secondFakeAddress).Return(fakeIommuGroup, nil).Times(1)
+		mockPCI.EXPECT().GetDeviceDriver(pciBasePath, secondFakeAddress).Return(fakeDriver, nil).Times(1)
+		mockPCI.EXPECT().GetDeviceNumaNode(pciBasePath, secondFakeAddress).Return(fakeNumaNode).Times(1)
+		mockPCI.EXPECT().GetDevicePCIID(pciBasePath, secondFakeAddress).Return(fakeID, nil).Times(1)
+		allowAnyDeviceCalls(*mockPCI)
+
+		iommuToPCIMap := make(map[string]string)
+		supportedPCIDeviceMap := make(map[string]string)
+		for _, pciDev := range fakePermittedHostDevices.PciHostDevices {
+			// do not add a device plugin for this resource if it's being provided via an external device plugin
+			if !pciDev.ExternalResourceProvider {
+				supportedPCIDeviceMap[pciDev.PCIVendorSelector] = pciDev.ResourceName
+			}
+		}
+
+		pciDevices := discoverPermittedHostPCIDevices(supportedPCIDeviceMap)
+		Expect(pciDevices[fakeName]).To(HaveLen(2), "two PCI device are expected to be found for: "+fakeName)
+		devs := constructDPIdevices(pciDevices[fakeName], iommuToPCIMap)
+		Expect(devs).To(HaveLen(2))
+		Expect(iommuToPCIMap).To(HaveLen(2))
+		devID := strings.Join([]string{fakeIommuGroup, fakeAddress}, deviceIDSeparator)
+		Expect(iommuToPCIMap[devID]).To(Equal(fakeAddress))
+		devSpec := formatVFIODeviceSpecs(devID)
+		Expect(devSpec).To(HaveLen(2))
+		Expect(devSpec[1].HostPath).To(Equal(filepath.Join(vfioDevicePath, fakeIommuGroup)))
```

Comment:
The first element in the slice returned by `formatVFIODeviceSpecs(devID)` is kind of static so testing it will not add any value to the test.
I changed to `Expect(devSpec).To(ContainElement(...)`